crates/proto/src/rr/rdata/caa.rs

```diff
@@ -40,7 +40,7 @@ use crate::{
 #[derive(Debug, PartialEq, Eq, Hash, Clone)]
 pub struct CAA {
     #[doc(hidden)]
-    pub issuer_critical: bool,
+    pub flags: u8,
```

Comment:
There's a bit of inconsistency amongst the record types with a flags field, but it seems like the most common approach is to:

 * store the flags individually in the record type (privately)
 * have a public flags() method to synthesize and return a u8/u16/etc containing all of the flags
 * have public flag_name() and set_flag_name(val) methods to get/set the individual flags.

Reply:
I'm not sure why that was marked `doc(hidden)` and not just kept private. My general preference is that we don't make interior structure part of the public API (i.e. keep the fields private).

In any case, I get why this is necessary, and since not all the flags are known to the code, preserving the reserved fields makes sense. To make sure we don't have too much overhead here, I would think keeping the `reserved_flags` is the correct thing (like the code is doing) and that isn't merged with the known booleans until we write it out. Which is pretty much what the code is doing.

Reply:
There's no setter for the reserved flags right now... so I think there's a behavioral question here, and that could lead to a confusing issue if the code has a setter for the reserved_flags. I'm not sure how to deal with that. maybe an assertion that if any of the fields we do understand are set, we'd fail in debug builds. and then zero out those fields when setting the reserved_flags.